examples/nlp/cai/generate_rl_cai_dataset.py

```diff
@@ -0,0 +1,741 @@
+import argparse
+import ast
+import json
+import os
+import os.path
+import random
+import re
+from pathlib import Path
+from typing import Dict, List, Optional, Union
+
+import numpy as np
+import requests
+from tqdm import tqdm
+
+
+class ChatPromptTemplate:
+    system_token = "System"
+    user_token = "User"
+    assistant_token = "Assistant"
+
+    system_turn_token = "<extra_id_0>"
+    turn_token = "<extra_id_1>"  # (model_config: chat_prompt_tokens.turn_start)
+    end_signal = "\n"  # (model_config: chat_prompt_tokens.end_of_turn. NOTE: "\x0A" is '\n' in ASCII code)
+    label_start = "<extra_id_2>"  # (model_config: chat_prompt_tokens.label_start)
+    end_name_signal = "\n"  # (model_config: chat_prompt_tokens.end_of_name. NOTE: "\x0A" is '\n' in ASCII code)
```

Comment:
Can we parametrize these and reuse the variable holding them throughout this script. `extra_id_*` is hardcoded in many places and it would be cleaner to reference a variable. WDYT about plumbing it from the parser?

Reply:
Thanks, done.
we have implemented the use of 'add_argument_group' to parameterize the variables.